modular_zzplurt/readme.md

```diff
@@ -0,0 +1,165 @@
+
+Our modularization protocols are founded on three pillars:
+
+1. **Modularize anything that's reasonable to modularize:** This includes anything that can be modularized without decreasing its quality or the quality of existing code, and won't logically cause more issues by modularizing than not doing it.
+2. **Use commenting conventions for non-modular edits:** When editing non-modular code, make sure to adequately use the commenting conventions to ensure it is known that it's a non-modular edit.
+3. **It's ok to mess up:** These rules are intended less as a set of guidelines for contributors to strictly follow, but rather coding standards that we all collectively help to attain, for examble through maintainers helping any contributors fulfill them in their pull requests.
+
+### Modular Overrides
+
+Modular overrides allow us to extend or modify the behavior of existing code without directly editing the core files. This is done by creating new definitions or modifying existing ones in a way that they can be easily integrated into the core codebase. There are two main types of modular overrides: Variable Overrides and Proc Overrides.
+
+#### Variable Overrides
+
+Variable overrides allow you to modify existing variables in a modular fashion. This is useful when you need to introduce new properties to existing objects without altering the core code. For example:
+
+```byond
+/obj/item/gun
+    var/muzzle_flash = TRUE
+
+    ...
+
+/obj/item/gun
+    muzzle_flash = FALSE
+```
+
+In this example, a new variable muzzle_flash is added to the /obj/item/gun object with a value of `TRUE`. The value of this variable is later changed to `FALSE` through overriding.
+
+#### Proc Overrides
+
+Proc overrides allow you to extend or modify the behavior of existing procedures (procs) without directly editing the core files. This is done by defining a new proc that calls the original proc and then adds the new behavior. For example:
+
+```byond
+/obj/item/gun/shoot_live_shot(mob/living/user, pointblank = 0, atom/pbtarget = null, message = 1)
+    . = ..() // Call the original proc and set its value to the default return value
+    if(muzzle_flash)
+        spawn_sparks(src) // Add new behavior
+```
+
+In this example, the shoot_live_shot proc is overridden to add a new behavior (spawning sparks) after calling the original proc. This ensures that the new behavior is added without modifying the core code.
+
```

Comment:
We should probably add something here (or somewhere else in the document, with a link) about file include order; proc overrides run in the order of last-defined to first-defined, and BYOND orders files in alphabetical order (with `_` being before all letters, and `~` being after). This would also potentially be a good spot to explain why the folder is `modular_zzplurt` rather than `modular_splurt`; `modular_zz` ensures that it comes after all the other `modular_` folders.

Reply:
Oh yeah that makes sense, it'd be a good idea to explain the order of including and compilation in byond. I think it'd be better to put it under this section